packages/recommend-react/src/useStatus.ts

```diff
@@ -0,0 +1,23 @@
+import { RecommendStatus } from '@algolia/recommend-vdom';
+import { useEffect, useRef, useState } from 'react';
```

Comment:
could we make a factory where useEffect, useRef, useState as passed to the hook creator, so we can share the implementation?

Reply:
Where would you see this implementation? If we have this duplication more frequently in the future, we could have a `recommend-hooks` package but right now it doesn't fit in `VDOM` or `core` IMO

Reply:
it could fit in vdom, seeing as only react-like api lives there at the moment. Whenever we would add Vue, I guess a different api than purely hooks would be needed, but not yet atm

Reply:
I think it's a bit too soon to abstract this, but we should definitely define a strategy if these duplications need to grow.